docs/writing-rules/data-flow/overview.md

```diff
@@ -23,4 +27,14 @@ Semgrep strives for simplicity and to deliver on the promise of lightweight and
 - No pointer or shape analysis: _Aliasing_ that happens in non-trivial ways may not be detected, such as through arrays or pointers. Individual elements in arrays or other data structures are not tracked. There is also no proper field sensitivity at present, but this may be developed in the future.
 - No soundness guarantees: Semgrep ignores the effects of `eval`-like functions on the program state. It doesn’t make worst-case sound assumptions, but rather "reasonable" ones.
 
-You can expect both false positives and false negatives. You should be able to remove false positives in different ways, for example, using [pattern-not](../../rule-syntax/#pattern-not) and [pattern-not-inside](../../rule-syntax/#pattern-not-inside). We definitely want to provide you with a way of eliminating false positives, so please [let file an issue if run into some](https://github.com/returntocorp/semgrep/issues/new/choose). We are rather happy to trade false negatives for simplicity and fewer false positives, but you are welcome to open a feature request if Semgrep misses some important bug you would like to catch, and we will certainly consider it.
+You can expect both false positives and false negatives. You can remove false positives in different ways, for example, using [pattern-not](../../rule-syntax/#pattern-not) and [pattern-not-inside](../../rule-syntax/#pattern-not-inside). We definitely want to provide you with a way of eliminating false positives, so please [let file an issue if run into some](https://github.com/returntocorp/semgrep/issues/new/choose). We are rather happy to trade false negatives for simplicity and fewer false positives, but you are welcome to open a feature request if Semgrep misses some important bug you would like to catch, and we will certainly consider it.
+
+## Data-flow status
+
+In principle, the data-flow analysis runs on any [language supported by](../../supported-languages.md).
+
+Internally, Semgrep translates the generic syntax tree into an analysis-friendly intermediate language, the _IL_, on which it runs _mostly_ language-agnostic analysis. However, this translation is not fully complete. Unsupported constructs are typically specific to one or a few languages.
```

Comment:
Not sure if users need to know this.

Reply:
Advanced users can appreciate this so I would keep it.

Reply:
Ok! Would like to understand how it works and also if you now in which direction to point me or recommend somebody who can explain... @IagoAbal

Reply:
You may be familiar with the term IR (Intermediate Representation) instead of IL, but the idea is Semgrep performs an analysis of the code, creating an abstract syntax tree, but apparently Semgrep's IL doesn't encode capture some features of some specific languages, which means there are some features in some languages that Semgrep simply ignores/doesn't support.

Reply:
@IagoAbal I would avoid making documentation only for a selected circle of enlightened users. What @efortuna explained makes sense. The minimal thing we should do is to spell out the abbreviations in their first occurrence at least...

Reply:
Ah ok, the abbreviation is expanded, just in a way I am not used to. Sorry. :D

Reply:
Resolving, commiting in: https://github.com/returntocorp/semgrep-docs/pull/694#discussion_r965957359